homeassistant/components/kostal_plenticore/switch.py

```diff
@@ -103,6 +116,94 @@ async def async_setup_entry(
             )
         )
 
+    # add excess AC energy switch if supported
+    if entry.data.get(CONF_SERVICE_CODE) is not None:
+        excess_energy_settings = await plenticore.client.get_setting_values(
+            "devices:local",
+            (
+                "EnergySensor:SensorPosition",
+                "Battery:Type",
+                "EnergySensor:InstalledSensor",
+            ),
+        )
+
+        try:
+            sensor_position = int(
+                excess_energy_settings["devices:local"]["EnergySensor:SensorPosition"]
+            )
+            battery_type = int(excess_energy_settings["devices:local"]["Battery:Type"])
+            installed_sensor = int(
+                excess_energy_settings["devices:local"]["EnergySensor:InstalledSensor"]
+            )
+        except (KeyError, ValueError):
+            _LOGGER.warning(
```

Comment:
Instead I would just make sure we have the right fields. When do we expect not to have these fields? We explicitly ask for them right?

Reply:
Agreed. removed the KeyError from the except block